app/Database/Migrations/2020-07-12-143952_create_monsters.php

```diff
@@ -0,0 +1,95 @@
+<?php namespace App\Database\Migrations;
+
+use CodeIgniter\Database\Migration;
+
+class CreateMonsters extends Migration
+{
+	public function up()
+	{
+		/**
+		 * Monsters
+		 *
+		 * Notice the `dungeon_id` field, a reference back to our `dungeons` table. This
+		 * implies that each monster "belongs to" a single dungeon, especially since it
+		 * cannot be null.
+		 */
+		$fields = [
+			'name'       => ['type' => 'varchar', 'constraint' => 255],
+			'health'     => ['type' => 'int', 'null' => true],
+			'dungeon_id' => ['type' => 'int'],
+			'created_at' => ['type' => 'datetime', 'null' => true],
+			'updated_at' => ['type' => 'datetime', 'null' => true],
+			'deleted_at' => ['type' => 'datetime', 'null' => true],
+		];
+		
+		$this->forge->addField('id');
+		$this->forge->addField($fields);
+
+		$this->forge->addKey('name');
+		$this->forge->addKey('created_at');
+		
+		$this->forge->createTable('monsters');
+
+		/**
+		 * Abilities
+		 *
+		 * Abilities may be "cast" by anything that has the ability. Below we will make
+		 * a separate table to define who "has" an ability, since different monsters
+		 * can have the same ability (or heroes, for that matter).
+		 */
+		$fields = [
+			'name'       => ['type' => 'varchar', 'constraint' => 255],
+			'damage'     => ['type' => 'int', 'null' => true],
+			'cooldown'   => ['type' => 'int', 'null' => true],
+			'created_at' => ['type' => 'datetime', 'null' => true],
+			'updated_at' => ['type' => 'datetime', 'null' => true],
+		];
+		
+		$this->forge->addField('id');
+		$this->forge->addField($fields);
+
+		$this->forge->addKey('name');
```

Comment:
Should this be a `uniqueKey`?

Reply:
It could be but I imagined "upgraded" abilities with the same name but higher stats. I think either way is fine, if you have a preference.

Reply:
Actually I like it being unique since we don't have that functionality many places, gives something else to play with.

Reply:
Oh, that makes sense. Just making sure it hadn't slipped by. Maybe on table with and one without just for example sakes?

Reply:
But I have no strong feelings either way :)